pkg/gui/commits_panel.go

```diff
@@ -16,19 +31,11 @@ func (gui *Gui) refreshCommits(g *gocui.Gui) error {
 			panic(err)
 		}
 		v.Clear()
-		red := color.New(color.FgRed)
-		yellow := color.New(color.FgYellow)
-		white := color.New(color.FgWhite)
-		shaColor := white
-		for _, commit := range gui.State.Commits {
-			if commit.Pushed {
-				shaColor = red
-			} else {
-				shaColor = yellow
-			}
-			shaColor.Fprint(v, commit.Sha+" ")
-			white.Fprintln(v, commit.Name)
+		displayStrings := make([]string, len(gui.State.Commits))
+		for i, commit := range gui.State.Commits {
+			displayStrings[i] = gui.renderCommit(commit)
```

Comment:
tried to generalise this by making an interface that contains the GetDisplayString() method but golang wasn't playing nicely with an array of interface types